apps/omg_watcher/lib/block_getter/core.ex

```diff
@@ -470,29 +466,24 @@ defmodule OMG.Watcher.BlockGetter.Core do
     cond do
       blknum_time == nil ->
         potential_block_withholdings = Map.put(potential_block_withholdings, blknum, %PotentialWithholding{time: time})
-
-        state = %{
-          state
-          | potential_block_withholdings: potential_block_withholdings
-        }
-
+        state = %{state | potential_block_withholdings: potential_block_withholdings}
         {:ok, state}
 
       time - blknum_time >= config.maximum_block_withholding_time_ms ->
         event = %Event.BlockWithholding{blknum: blknum, hash: hash}
-
-        state =
-          state
-          |> set_chain_status(:error)
-          |> add_non_existing_event(event)
-
+        state = state |> set_chain_status(:error) |> add_distinct_event(event)
         {{:error, :withholding}, state}
 
       true ->
         {:ok, state}
     end
   end
 
+  defp not_queued_up_yet?(number, unapplied_blocks), do: not Map.has_key?(unapplied_blocks, number)
+
```

Comment:
Maybe it will be more readable to introduce following modification:
```
defp queued_up_yet?(number, unapplied_blocks), do: Map.has_key?(unapplied_blocks, number)
```

Reply:
and then: `not queued_up_yet()`

Reply:
ok I dug into this and realized I'd rather do more massive renames into the `block_getter` area, so I'll add a chore.

For now I'll keep as is - the idea is to have a specific and encompassing function and just a simplest `true <- specific_function() || :error` construction in the with. Removing the confusion that `not_` prefix brings in is probably a matter of the larger renaming